### PR TITLE
horizon: Publish verify-range image in master branch

### LINKED ISSUE #2761: Improve verify-range testing
While testing captive core with verify-range in AWS batch I encountered a few problems which made the experience unpleasant:

1. Manually editing the Job Definitions/Queues/Compute environments in the AWS console was hairy and error-prone. 
2. Spawning jobs suffers from (1) aggravated by the need to calculate the ledger ranges (we have a spreadsheet for that, but still)
3. When a job fails, there is no way to simply respawn jobs. You need to manually create a new job by copy-pasting the fields of the failed job (increasing the number of retries is often not enough).

To improve the situation, I propose the following:
1. Use [Terraform to keep track of the definiton AWS Batch resources](https://www.terraform.io/docs/providers/aws/r/batch_compute_environment.html) and apply them automatically on merge. This will probably require having a separate repo for verify-range.
2. Spawn test jobs semi-automatically through CI (specify the base branch, click a button and have CI spawn the jobs). For GitHub Actions we could use https://www.actionspanel.app/ (I don't know if there is an equivalent solution for CircleCI)
3. CI could monitor jobs for failures and respawn them when the job failure is transient. We could also have a manual CI action which clones a job given its identifier.

### LINKED ISSUE #3796: Automatically publish verify-range image to docker hub
`stellar/verify-range` image can now be published to docker hub similarly to `stellar/horizon`.

### PR DESCRIPTION
**Pull request by 2opremio**
Fixes #2761 
Fixes #3796

### CODE REVIEW

**Review by bartekn — COMMENTED** [review]
There's a actually a lot of code duplication. Maybe we could use params (see `enable-captive-core` param)? Then if, say, `publish-image` is `true` we just run `docker publish` command, everything else stays the same.

File: .circleci/config.yml [hunk 1]
```diff
@@ -471,12 +478,17 @@ workflows:
       - test_horizon_integration:
           name: test_horizon_integration_with_captive_core
           enable-captive-core: true
-      - test_verify_range_docker_image
+      - test_verify_range_docker_image:
+          # test pusblishing
+          publish: true
```
Comment (bartekn) [inline .circleci/config.yml:483]: We don't want to publish on non-master, correct?
```suggestion
          publish: false
```

**Review by bartekn — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (2opremio) [reply .circleci/config.yml:483]: I'm testing that it works before merging

Comment (2opremio) [reply .circleci/config.yml:483]: But ... it seems like the secrets are not there (probably due to this being a PR from my fork).

I will revert it and test from `master`

**Comment by 2opremio** [comment]
@bartekn PTAL

**Review by bartekn — APPROVED** [review]

**Pull request merged**